src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -108,22 +108,13 @@ internal void SetEntry(CacheEntry entry)
 
             DateTimeOffset utcNow = _options.Clock.UtcNow;
 
-            DateTimeOffset? absoluteExpiration = null;
-            if (entry.AbsoluteExpirationRelativeToNow.HasValue)
-            {
-                absoluteExpiration = utcNow + entry.AbsoluteExpirationRelativeToNow;
-            }
-            else if (entry.AbsoluteExpiration.HasValue)
-            {
-                absoluteExpiration = entry.AbsoluteExpiration;
-            }
-
             // Applying the option's absolute expiration only if it's not already smaller.
             // This can be the case if a dependent cache entry has a smaller value, and
             // it was set by cascading it to its parent.
-            if (absoluteExpiration.HasValue)
+            if (entry.AbsoluteExpirationRelativeToNow.HasValue)
             {
-                if (!entry.AbsoluteExpiration.HasValue || absoluteExpiration.Value < entry.AbsoluteExpiration.Value)
+                var absoluteExpiration = utcNow + entry.AbsoluteExpirationRelativeToNow.Value;
```

Comment:
Since this library also ships in a .NET Standard flavor, it may be better to use `Nullable<T>.GetValueOrDefault()` instead of `Nullable<T>.Value` when  `Nullable<T>.HasValue` returned **true**. 

In .NET 6.0, the JIT was optimized to handle this (see @stephentoub's great blog post). The JIT in older versions of the .NET (and .NET Framework) does not have this optimization though.

Reply:
Let's do this in a follow up PR, if necessary.